runtime/near-vm-logic/src/gas_counter.rs

```diff
@@ -22,6 +22,8 @@ pub struct GasCounter {
     prepaid_gas: Gas,
     is_view: bool,
     ext_costs_config: ExtCostsConfig,
+    /// Where to store profile data, if needed.
+    profile: Option<ProfileData>,
```

Comment:
Instead of put it into the constructor, can you put it using `setProfile` method. This way it can be behind a feature and doesn't affect existing implementations.

Reply:
But is it really correct for the cost breakdown be behind the flag?